db-queue-spring/src/test/java/ru/yoomoney/tech/dbqueue/spring/dao/CustomH2PickTaskDaoTest.java

```diff
@@ -0,0 +1,101 @@
+package ru.yoomoney.tech.dbqueue.spring.dao;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import ru.yoomoney.tech.dbqueue.api.EnqueueParams;
+import ru.yoomoney.tech.dbqueue.api.TaskRecord;
+import ru.yoomoney.tech.dbqueue.dao.PickTaskSettings;
+import ru.yoomoney.tech.dbqueue.dao.QueuePickTaskDao;
+import ru.yoomoney.tech.dbqueue.settings.QueueLocation;
+import ru.yoomoney.tech.dbqueue.settings.TaskRetryType;
+import ru.yoomoney.tech.dbqueue.spring.dao.utils.H2DatabaseInitializer;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.hamcrest.CoreMatchers.*;
+
+public class CustomH2PickTaskDaoTest extends QueuePickTaskDaoTest {
+
+    @BeforeClass
+    public static void beforeClass() {
+        H2DatabaseInitializer.initialize();
+    }
+
+    public CustomH2PickTaskDaoTest() {
+        super(
+                new H2QueueDao(
+                        H2DatabaseInitializer.getJdbcTemplate(),
+                        H2DatabaseInitializer.CUSTOM_SCHEMA),
+                pickTaskSettings ->
+                        new H2QueuePickTaskDao(
+                                H2DatabaseInitializer.getJdbcTemplate(),
+                                H2DatabaseInitializer.CUSTOM_SCHEMA,
+                                pickTaskSettings,
+                                H2DatabaseInitializer.getTransactionTemplate()),
+                H2DatabaseInitializer.CUSTOM_TABLE_NAME,
+                H2DatabaseInitializer.CUSTOM_SCHEMA,
+                H2DatabaseInitializer.getJdbcTemplate(),
+                H2DatabaseInitializer.getTransactionTemplate());
+    }
+
+    @Override
+    protected String currentTimeSql() {
+        return "now()";
+    }
+
+
+    @Test
+    public void pick_task_concurrently() throws Exception {
```

Comment:
What do you expect from this test? It won't fail even in the previous version of implementation

Reply:
It didn't fail cause in the previous implementation `for update` cause was used and the database blocked rows. In the current version, it doesn't do it, so there aren't these strict guarantees.   
The test should check to pick up all tasks from the queue in the multithreaded environment in the right way.